### PR TITLE
AuditHandler refactor

### PR DESCRIPTION
**Pull request by labkey-matthewb**
#### Rationale
Consolidate audit event generation in AuditHandler.  Remove redundant TableInfo and QueryService methods.
Also, incorporate the change to enable detailed audit logging for dataset.

#### Related Pull Requests
https://github.com/LabKey/platform/pull/1935
https://github.com/LabKey/inventory/pull/181

### CODE REVIEW

File: api/src/org/labkey/api/data/TableInfo.java [hunk 1]
```diff
@@ -249,7 +258,7 @@ default String getDbSequenceName(String columnName)
      * Return the default query grid view URL for the table or null.
      * Instead of calling this method directly, callers should pass
      * {@link QueryAction#executeQuery} to
-     * {@link org.labkey.api.query.QueryView#urlFor(QueryAction)} or
+     * {@link org.labkey.api.query.QueryView#cd serurlFor(QueryAction)} or
```
Comment (labkey-susanh) [inline api/src/org/labkey/api/data/TableInfo.java]: This looks unintentional

File: api/src/org/labkey/api/audit/AuditHandler.java [hunk 2]
```diff
@@ -20,143 +22,173 @@
...
+                       @Nullable List<Map<String, Object>> rows, @Nullable List<Map<String, Object>> existingRows);
 
-    /**
-     * Allow for adding fields that may be present in the updated row but not represented in the original row
-     * @param originalRow the original data
-     * @param modifiedRow the data from the updated row that has changed
-     * @param updatedRow the row that has been updated, which may include fields that have not changed
-     */
-    protected void addDetailedModifiedFields(Map<String, Object> originalRow, Map<String, Object> modifiedRow, Map<String, Object> updatedRow)
+    abstract class AbstractAuditHandler implements AuditHandler
```
Comment (labkey-susanh) [inline api/src/org/labkey/api/audit/AuditHandler.java:36]: I don't know that I like the pattern of having the Abstract class inside the Interface, but that's probably just me.

**Review by labkey-susanh — APPROVED** [review]
Seems fine as long as our timeline tests an location history tests are happy in Sample Manager and Inventory.

File: api/src/org/labkey/api/audit/AuditHandler.java (hunk 2 above)
Comment (labkey-matthewb) [reply api/src/org/labkey/api/audit/AuditHandler.java:36]: Not just you.  I was just trying to minimize the diff for now.  More to come...

File: api/src/org/labkey/api/data/TableInfo.java (hunk 1 above)
Comment (labkey-matthewb) [reply api/src/org/labkey/api/data/TableInfo.java]: thanks

**Pull request merged**